pkg/config/token_signing_config.go

```diff
@@ -28,76 +28,21 @@ type TokenSigningConfig struct {
 	// configuration.
 	Keys keys.Config `env:", prefix=TOKEN_"`
 
-	// TokenSigningKeys is the parent token signing key (not the actual signing
-	// version). It is an array for backwards-compatibility, but in practice it
-	// should only have one element.
-	//
-	// Previously it was a list of all possible signing key versions, but those
-	// have moved into the database.
-	//
-	// TODO(sethvargo): Convert to string in 0.22+.
-	TokenSigningKeys []string `env:"TOKEN_SIGNING_KEY, required"`
-
-	// TokenSigningKeyIDs specifies the list of kids, corresponding to the
-	// TokenSigningKey
-	//
-	// TODO(sethvargo): Remove in 0.22+.
-	//
-	// Deprecated: moved into the database.
-	TokenSigningKeyIDs []string `env:"TOKEN_SIGNING_KEY_ID, default=v1"`
+	// TokenSigningKey is the parent token signing key (not the actual signing
+	// version).
+	TokenSigningKey string `env:"TOKEN_SIGNING_KEY, required"`
 
 	// TokenIssuer is the `iss` field on the JWT.
 	TokenIssuer string `env:"TOKEN_ISSUER, default=diagnosis-verification-example"`
 }
 
-// ParentKeyName returns the name of the parent key.
-func (t *TokenSigningConfig) ParentKeyName() string {
-	// Validation prevents this slice from being empty.
-	value := t.TokenSigningKeys[0]
-
-	// This is Google-specific, but that's the only platform where we can
-	// meaningfully detect this.
-	if idx := strings.Index(t.TokenSigningKeys[0], "/cryptoKeyVersions"); idx != -1 {
```

Comment:
Should we move this to validation and fail if we detect that substring now?

Reply:
yes - I think so

Reply:
Done